packages/core/src/runtime/Test.ts

```diff
@@ -184,16 +181,19 @@ export default class Test implements ITest {
 			}
 
 			debug('running hook function: beforeAll')
-			await this.runHookFn(this.hook.beforeAll, browser, testDataRecord)
+			await this.runHookFn(
+				this.hook.beforeAll,
+				browser,
+				testDataRecord,
+				HookType.beforeAll,
+				testObserver,
+			)
```

Comment:
Just a sugesstion, I think we do not need arg `HookType.beforeAll`, can we get the `type` from the `this.hook.beforeAll` by adding property in Hookbase? 
```suggestion
			await this.runHookFn(
				this.hook.beforeAll,
				browser,
				testDataRecord,
				testObserver,
			)
```